src/hooks/tls.ts

```diff
@@ -0,0 +1,48 @@
+import { Options } from 'got-cjs';
+// @ts-expect-error Missing types
+import { getBrowser, getUserAgent } from 'header-generator/src/utils';
+import { ciphers, sigalgs, minVersion, maxVersion } from '../scraping-defaults';
+
+type Browser = 'chrome' | 'firefox' | 'safari' | undefined;
+
+export function tlsHook(options: Options): void {
+    const { https } = options;
+
+    if (https.ciphers || https.signatureAlgorithms || https.minVersion || https.maxVersion) {
+        return;
+    }
+
+    const browser: Browser = getBrowser(getUserAgent(options.headers));
+
+    const useDefault = () => {
+        https.ciphers = ciphers.firefox;
+        https.signatureAlgorithms = sigalgs.firefox;
+        https.minVersion = minVersion.firefox;
+        https.maxVersion = maxVersion.firefox;
+    };
+
+    if (browser) {
+        if (browser in ciphers) {
+            // This is ugly because TS doesn't type object[nonExistent] as undefined
+            https.ciphers = ciphers[browser as keyof typeof ciphers];
+            https.signatureAlgorithms = sigalgs[browser as keyof typeof ciphers];
+            // @ts-expect-error @types/node doesn't accept TLSv1.0
+            https.minVersion = minVersion[browser as keyof typeof ciphers];
+            https.maxVersion = maxVersion[browser as keyof typeof ciphers];
+
+            return;
+        }
+
+        if (browser === 'safari') {
+            https.ciphers = ciphers.chrome;
+            https.signatureAlgorithms = sigalgs.chrome;
+            // @ts-expect-error @types/node doesn't accept TLSv1.0
+            https.minVersion = minVersion.chrome;
+            https.maxVersion = maxVersion.chrome;
+
+            return;
+        }
+    }
+
+    useDefault();
```

Comment:
No because these `if`s are inside another `if`. We'd have to duplicate `useDefault()` calls.

Reply:
```js
if (browser in ciphers) {
    // if browser is falsy, it can't be in ciphers because ciphers are never "", false, 0, null, undefined, NaN
} else if (browser === 'safari') {
    // if browser is falsy, it can't be safari
} else {
    useDefault();
}
```

Am I blind again?

Reply:
I don't wanna argue about code style so I'll leave this up to you